File: TCM App/Models/Training.cs
using TCM_App.Models.Enums;

namespace TCM_App.Models
{
    public class Training
    {
        public int Id { get; set; }
        public required DateTime Date { get; set; }
        public required string Description { get; set; }

        //This will be the coach
        public int MemberId { get; set; }
        public Member Member { get; set; } = null!; // Ensure Member is not null

        public int Status { get; set; }


        public int ClubId { get; set; }
        public Club Club { get; set; } = null!; // Ensure Club is not null
        public List<MemberTraining> MemberTrainings { get; set; } = []; // Initialize to an empty list

        public TrainingType TrainingType { get; set; } 
    }
}

File: TCM_Client/src/app/_models/Training.ts
import {
  MemberTrainingData,
  SimpleMemberAttendanceData,
} from './MemberTrainingData';

export interface Training {
  id: number;
  date: Date;
  description: string;
  status: string;
  notes?: string;
  trainingType: string;
}

export interface TrainingDetails {
  id: number;
  date: Date;
  description: string;
  status: string;
  trainingType: string;
  memberTrainings: MemberTrainingData[];
}

export interface TrainingForEdit {
  id: number;
  date: Date;
  description: string;
  status: string;
  trainingType: string;
  membersToAttend: SimpleMemberAttendanceData[];
}
